Explore: Interaction tracking for logs pinning and filtering in content outline

harisrozajac opened the pull request:
This PR adds interaction tracking for logs pinning and filtering in the content outline. Logs pinning was added in #88128 and logs filtering in #86431. 

Thanos has approved these new interaction events. We agreed to just add new type values which will be consistent with the event type we already have for content outline in rudderstack. 

**Which issue(s) does this PR fix?**:

Closes #88730 

**Special notes for your reviewer:**

Please check that:
- [ ] Each interaction in the UX logs a corresponding interaction defined in this PR.

harisrozajac committed:
Add interaction tracking

ifrost reviewed (approved):
LGTM 🎉 

Just wonder if moving `reportInteraction` out of scrollIntoView could improve readability a bit, something like:

```typescript
  const handleItemClicked = (child: ContentOutlineItemContextProps) => {
    scrollIntoView(child.ref, child.panelId, child.type, child.customTopOffset)
    if (child.type === 'filter') {
      activateFilter(child.id);
    } else {
      reportInteraction('explore_toolbar_contentoutline_clicked', {
        item: 'select_section',
        type: child.panelId,
      });      
    }
  }
```

harisrozajac committed:
Move reportInteraction out of scrollIntoView to improve readability

harisrozajac commented:
> Just wonder if moving `reportInteraction` out of scrollIntoView could improve readability a bit

Thanks for this suggestion. I've implemented it and removed the onClick logic inside child item as `handleClickItemClick` covers it.

gtk-grafana reviewed (approved):
Nice! LGTM

harisrozajac committed:
move activateFilter logic inside handleItemClicked
